web/src/recent_view_ui.js

```diff
@@ -1553,9 +1553,18 @@ export function initialize({
         }, 300),
     );
 
+    $("body").on("input", "#recent_view_search", () => {
+        if ($("#recent_view_search").val()) {
+            $("#recent_view_search_clear").css("visibility", "visible");
+        } else {
+            $("#recent_view_search_clear").css("visibility", "hidden");
+        }
```

Comment:
Let's use `display` instead of `visibility`.

Reply:
I read that zulip has a policy of not doing .css() instead of classes unless absolutely necessary but here flex properties were growing input fields(search bars)unnecessarily when there is no "x" button while using .hide() / .show(). Can you tell how to proceed?

Reply:
You need to use `position: fixed` for the the clear search icons.

Reply:
> You need to use `position: fixed` for the the clear search icons.

I tried different solutions and now it's working with ```position: absolute``` for 'x' button. Kindly review.